nix/checks/monitor.nix

```diff
@@ -14,29 +13,27 @@
     scale-network.services.monitoring.enable = true;
   };
 
-  nodes.client1 =
-    { pkgs, ... }:
-    {
-      systemd.services.systemd-networkd.environment.SYSTEMD_LOG_LEVEL = "debug";
-      environment = {
-        systemPackages = with pkgs; [
-          curl
-        ];
-      };
+  nodes.client1 = {pkgs, ...}: {
+    systemd.services.systemd-networkd.environment.SYSTEMD_LOG_LEVEL = "debug";
+    environment = {
+      systemPackages = with pkgs; [
+        curl
+      ];
     };
+  };
+
+  testScript = {nodes, ...}: ''
+    start_all()
+    coremaster.succeed("sleep 2")
+    coremaster.wait_for_unit("grafana.service", None, 30)
+    coremaster.wait_for_unit("prometheus.service", None, 30)
+    coremaster.wait_until_succeeds("nc -vz localhost 3000")
 
-  testScript =
-    { nodes, ... }:
-    ''
-      start_all()
-      coremaster.succeed("sleep 2")
-      coremaster.wait_for_unit("grafana.service", None, 30)
-      coremaster.wait_for_unit("prometheus.service", None, 30)
-      coremaster.wait_until_succeeds("nc -vz localhost 3000")
+    client1.wait_until_succeeds("ping -c 5 ${nodes.coremaster.networking.hostName}")
+    client1.wait_until_succeeds("curl -v -L -H \"Host: monitoring.scale.lan\" http://${nodes.coremaster.networking.hostName}")
```

Comment:
From what I have found, NixOS VM testing overrides the hostname with the node name defined in the test config and unfortunately you can not have `.` in the name. So in this case you would have to replace the first one with:

```
client1.wait_until_succeeds("ping -c 5 coremaster")
```

And the second check won't work since the Nginx proxy was removed.